rest/model/createhost.go

```diff
@@ -9,15 +12,18 @@ type CreateHost struct {
 
 func (createHost *CreateHost) BuildFromService(h interface{}) error {
 	switch v := h.(type) {
-	case *CreateHost:
-		createHost.DNSName = v.DNSName
-		createHost.InstanceID = v.InstanceID
+	case host.Host:
+		createHost.DNSName = v.Host
+		createHost.InstanceID = v.ExternalIdentifier
+	case *host.Host:
```

Comment:
I think you can just do `case host.Host, *host.Host:`

Reply:
You can't, because if you do it's an interface, and you can't access its fields.